tests/test_logger.py

```diff
@@ -0,0 +1,84 @@
+"""
+Tests for the ralph.logger module
+"""
+
+import pytest
+from click.testing import CliRunner
+
+import ralph.logger
+from ralph.cli import cli
+from ralph.exceptions import ConfigurationException
+
+
+# pylint: disable=invalid-name, unused-argument
+def test_logger_exists(fs, monkeypatch):
+    """Test the logging system when a correct configuration is provided"""
+
+    mock_default_config = {
+        "version": 1,
+        "propagate": True,
+        "formatters": {
+            "ralph_format": {
+                "format": "%(asctime)-23s %(levelname)-8s %(name)-8s %(message)s"
+            },
+        },
+        "handlers": {
+            "ralph_stderr": {
+                "class": "logging.StreamHandler",
+                "level": "INFO",
+                "stream": "ext://sys.stderr",
+                "formatter": "ralph_format",
+            },
+        },
+        "loggers": {
+            "ralph": {
+                "handlers": ["ralph_stderr"],
+                "level": "INFO",
+            },
+        },
+    }
+
+    fs.create_dir("/dev")
+
+    monkeypatch.setattr(ralph.logger, "LOGGING_CONFIG", mock_default_config)
+
+    runner = CliRunner()
+    result = runner.invoke(
+        cli,
+        ["push", "-b", "fs", "test_file"],
+        input="test input",
+    )
+
+    assert result.exit_code == 0
+
+    assert "Pushing archive test_file to the configured fs backend" in result.output
+
+
+# pylint: disable=invalid-name, unused-argument
+def test_logger_no_config(fs, monkeypatch):
+    """Test that an error occurs when no logging configuration exists"""
+
+    mock_default_config = None
+
+    monkeypatch.setattr(ralph.logger, "LOGGING_CONFIG", mock_default_config)
+
+    runner = CliRunner()
+
+    with pytest.raises(ConfigurationException):
+        result = runner.invoke(cli, ["list", "-b", "fs"], catch_exceptions=False)
+        assert result.exit_code == 1
```

Comment:
```suggestion
    assert result.exit_code == 1
```

Reply:
This is not possible since it generates an error: 'result' referenced before assignment

Reply:
arf :sweat_smile: